source_modelling/sources.py

```diff
@@ -151,6 +152,47 @@ def wgs_depth_coordinates_to_fault_coordinates(
             return np.array([1 / 2, 1 / 2])  # Point is in the centre of the small patch
         raise ValueError("Given global coordinates out of bounds for point source.")
 
+    def rrup_distance(self, point: np.ndarray) -> float:
+        """Compute RRup Distance between a fault and a point.
+
+        Parameters
+        ----------
+        point : np.ndarray
+            The point to compute distance to (in lat, lon, depth format)
+
+
+        Returns
+        -------
+        float
+            The rrup distance (in metres) between the point and the fault geometry.
+
+        Raises
+        ------
+        ValueError
+            If the optimisation routine fails to converge for the point.
```

Comment:
No optimization done in the case of a Point?

Reply:
Good catch.